docs/technical-documentation/rest-create.md

```diff
@@ -0,0 +1,197 @@
+
+## Files and Media
+
+Drupal is supposed to have a way to upload files, but this seems to require the use of an X-CSRF-Token, which can only be retrieved using Cookie authentication and even then does not allow you to upload.
+
+Consequently there is a REST endpoint not listed in the REST UI, because it is not configurable and is part of the Islandora 8 system.
+
+This endpoint is available at `http://localhost:8000/node/{node id}/media/{media type}/{media use}`
+
+It only accepts PUT requests. If the media and file don't exist they are created, if they exist the file is updated with the new body.
+
+The node and taxonomy term are used to search (via an [entity query](https://api.drupal.org/api/drupal/core!lib!Drupal.php/function/Drupal%3A%3AentityQuery/8.6.x)) for a media. If this media exists the body of the file is replaced with the new content, otherwise a new file is created to hold the contents.
+
+The tokens to this URI are as follows:
+
+1. node id : The numeric ID of the node you wish to link this media/file to.
+1. media type : The media type name you wish to create (ie. image, file, audio)
+1. media use : The numeric ID of the media use taxonomy term to set for this media 
+
+You can find the media use taxonomy terms at `http://localhost:8000/admin/structure/taxonomy/manage/islandora_media_use/overview`
+
+So the body of the request is the actual binary file to upload.
+
+&#x1F34E; For example: 
+
+With a local file called `my-image.png` that I wanted to link to a node with ID `3`.
+
+I am using the taxonomy term "Original file", which on my machine is `16`
+
+```
+> curl -i -X PUT -u admin:islandora -H"Content-type: image/png" --data-binary "@my-image.png" -H"Content-Location: public://images/my-image.png" 'http://localhost:8000/node/3/media/image/16'
+
+HTTP/1.1 100 Continue
+
+HTTP/1.1 201 Created
+Date: Tue, 05 Mar 2019 22:01:39 GMT
+Server: Apache/2.4.18 (Ubuntu)
+X-Powered-By: PHP/7.1.26-1+ubuntu16.04.1+deb.sury.org+1
+Cache-Control: must-revalidate, no-cache, private
+Location: http://localhost:8000/media/4
+X-UA-Compatible: IE=edge
+Content-language: en
+X-Content-Type-Options: nosniff
+X-Frame-Options: SAMEORIGIN
+Expires: Sun, 19 Nov 1978 05:00:00 GMT
+Vary: 
+X-Generator: Drupal 8 (https://www.drupal.org)
+Content-Length: 0
+Content-Type: text/html; charset=UTF-8
+```
+
+The parts of the above request are:
+
+1. `-i` - return the response headers 
+1. `-X PUT` - send a PUT request
+1. `-u admin:islandora` - use these basic authentication credentials
+1. `-H"Content-type: image/png"` - send the content-type header
+1. `--data-binary "@my-image.png"` - send the contents of the file located at my-image.png as binary
+1. `-H"Content-Location: public://images/my-image.png"` - store the file in the public scheme at the path `images/my-image.png`
+1. `'http://localhost:8000/node/3/media/image/16'` - the endpoint of the request specifying the node, media type and taxonomy term.
+
```

Comment:
Suggest adding an example where the file is persisted to Fedora:

"If you want to persist the file to Fedora, use a `-H"Content-Location` header with a value like `fedora://my-image.png`.